src/cloupe/cloupe.py

```diff
@@ -232,19 +232,41 @@ def annotations_writer(self):
             )
 
     # plotting the spatial info
-    def spatial_plot(self):
+    def spatial_projection(self):
+        spatial_plot = self.cwd + '/projection_spatial.csv'
         spatial_embedding = self.projections['Spatial']
-        plt.scatter(x=spatial_embedding[0], y=spatial_embedding[1])
-        plt.gca().invert_yaxis()
-        plt.show()
+        projections_1 = [["Barcodes"]+self.matrices[0]["Barcodes"]]
+        for column in range(len(spatial_embedding)):
+            projections_1.append(["d{}".format(column)]+[element for element in spatial_embedding[column]])
+        transposed_projections_1 = zip(*projections_1)
+        with open(spatial_plot, "w", newline="") as spatial_projection_file:
+            csv_writer = csv.writer(spatial_projection_file)
+            csv_writer.writerows(transposed_projections_1)
+
+        # plt.scatter(x=spatial_embedding[0], y=spatial_embedding[1])
+        # plt.gca().invert_yaxis()
+        # plt.show()
+
+    # plotting the tsne info
+    def tsne_projection(self):
+        tsne_plot = self.cwd + '/projection_tsne.csv'
+        tsne_embedding = self.projections['tsne']
+        projections_2 = [["Barcodes"] + self.matrices[0]["Barcodes"]]
+        for column in range(len(tsne_embedding)):
+            projections_2.append(["d{}".format(column)] + [element for element in tsne_embedding[column]])
+        transposed_projections_2 = zip(*projections_2)
+        with open(tsne_plot, "w", newline="") as tsne_projection_file:
+            csv_writer = csv.writer(tsne_projection_file)
+            csv_writer.writerows(transposed_projections_2)
```

Comment:
Don't do `Spatial` and `tsne` as separate functions do a single function that goes over all the available projections. Something along the lines of :
```python
projections = {}
# loop through all the avaialble projections
for projection_name in self.projections.keys():
    projections[projection_name] = []
    # process each column for this projection
    for column in range(len(self.projections[projection_name])):
            projections[projection_name].append(["d{}".format(column)]+[element for element in self.projections[projection_name][column]])
    # write csv for this one use projection_name variable as part of the name
    with open(f"projection_{projection_name}.csv", "w") as f:
            csv_writer = csv.writer(f)
            csv_writer.writerows(projections[projection_name])

Reply:
Done